dashboard/lib/user_permission_concern.rb

```diff
@@ -0,0 +1,98 @@
+require 'cdo/chat_client'
+
+#
+# Module to mix-in to User model providing expressive interactions with
+# user permissions.
+#
+module UserPermissionConcern
```

Comment:
Any particular reason why this is under `dashboard/lib` instead of the usual `dashboard/app/model/concerns`?

Reply:
nit: our existing [model concerns](https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/app/models/concerns), and the best I can tell from online guidance (though that is sparse. here's an [example](https://medium.freecodecamp.org/add-callbacks-to-a-concern-in-ruby-on-rails-ef1a8d26e7ab)) is to not use the name `Concern` in the class name. Maybe something like `Permissible` or `WithPermissions`? I'm not too hung up on this though. I'm also ok with `UserPermissionConcern` especially if this is just meant to split up `user.rb` and not intended to be shared with other models.

Reply:
:joy: Because I didn't know that directory existed.  TIL!

I did get a little stuck on the name.  `Permissible` feels like it describes a behavior rather than a model, `WithPermissions` I like better but "With*" doesn't seem to be a pattern for our concerns as yet.  `include HasPermissions` sort of has the same problem but it might read better as a concern.  `include UserPermissionGrantee` is verbose but at least it reads like an interface name.

Reply:
Q: Should the fairly new `UserMultiAuthHelper` also live in `dashboard/app/model/concerns` even though it's not extending `Concern` yet?  It also exists to pull some methods out of User.

Reply:
I think only actual Rails concerns (extending `Concern`) should go in that directory for clarity. Other helper modules and classes can always live in `dashboard/lib`.